src/engraving/rendering/dev/layoutindepended.cpp

```diff
@@ -0,0 +1,228 @@
+    {
+        switch (type) {
+        case AccidentalBracket::PARENTHESIS: return { SymId::accidentalParensLeft, SymId::accidentalParensRight };
+        case AccidentalBracket::BRACKET: return { SymId::accidentalBracketLeft, SymId::accidentalBracketRight };
+        case AccidentalBracket::BRACE: return { SymId::accidentalCombiningOpenCurlyBrace, SymId::accidentalCombiningCloseCurlyBrace };
+        case AccidentalBracket::NONE: return { SymId::noSym, SymId::noSym };
+        }
+        return { SymId::noSym, SymId::noSym };
+    };
+
+    // Single?
+    SymId singleSym = accidentalSingleSym(item);
+    if (singleSym != SymId::noSym && conf.engravingFont()->isValid(singleSym)) {
+        Accidental::LayoutData::Sym s(singleSym, 0.0, 0.0);
+        ldata->syms.push_back(s);
+
+        ldata->addBbox(item->symBbox(singleSym));
+    }
+    // Multi
+    else {
+        double margin = conf.styleMM(Sid::bracketedAccidentalPadding);
+        double x = 0.0;
+
+        std::pair<SymId, SymId> bracketSyms;
+        bool isNeedBracket = item->bracket() != AccidentalBracket::NONE && !item->parentNoteHasParentheses();
+        if (isNeedBracket) {
+            bracketSyms = accidentalBracketSyms(item->bracket());
+        }
+
+        // Left
+        if (bracketSyms.first != SymId::noSym) {
+            Accidental::LayoutData::Sym ls(bracketSyms.first, 0.0,
+                                           item->bracket() == AccidentalBracket::BRACE ? item->spatium() * 0.4 : 0.0);
+            ldata->syms.push_back(ls);
+            ldata->addBbox(item->symBbox(bracketSyms.first));
+
+            x += item->symAdvance(bracketSyms.first) + margin;
+        }
+
+        // Main
+        SymId mainSym = item->symId();
+        Accidental::LayoutData::Sym ms(mainSym, x, 0.0);
+        ldata->syms.push_back(ms);
+        ldata->addBbox(item->symBbox(mainSym).translated(x, 0.0));
+
+        // Right
+        if (bracketSyms.second != SymId::noSym) {
+            x += item->symAdvance(mainSym) + margin;
+
+            Accidental::LayoutData::Sym rs(bracketSyms.second, x,
+                                           item->bracket() == AccidentalBracket::BRACE ? item->spatium() * 0.4 : 0.0);
+            ldata->syms.push_back(rs);
+            ldata->addBbox(item->symBbox(bracketSyms.second).translated(x, 0.0));
+        }
+    }
+}
+
+void LayoutIndepended::layout(const ActionIcon* item, ActionIcon::LayoutData* ldata)
+{
+    FontMetrics fontMetrics(item->iconFont());
+    ldata->setBbox(fontMetrics.boundingRect(Char(item->icon())));
+}
+
+void LayoutIndepended::layout(const Articulation* item, Articulation::LayoutData* ldata)
+{
+    if (item->isHiddenOnTabStaff()) {
+        ldata->setIsSkipDraw(true);
+        return;
+    }
+    ldata->setIsSkipDraw(false);
+
+    RectF bbox;
+
+    if (item->textType() == ArticulationTextType::NO_TEXT) {
+        bbox = item->symBbox(item->symId());
```

Comment:
Here we use the symId, so we assume this has already been computed. But that is computed based on whether the articulation is above or below, and that info depends on whether the chord is up or down... so I'm afraid we can't layout Articulation independently. 

Actually we already see here, that it is difficult to keep track of dependencies this way. Everything with that regard needs to be done manually, and a lot of the knowledge about it is only in our heads. 
Perhaps we should still think about some kind of dependency system...

Reply:
I added update `bbox` on change `symId`
We can say `independent` if the layout only depends on its own properties. And then we can add similar updates like I just did.

Reply:
It turns out an interesting point, we do not need to subscribe to all item properties, from which the layout data depends. We need to subscribe only to those properties that can be changed during the layout. Because at the beginning of the layout, we reset the layout data and recalculate it. Maybe in the future we can not reset it between calls, then we need to subscribe to changes in all of item properties on which it depends